### PR TITLE
Lets RCDs construct floors on lattices/catwalks

### LINKED ISSUE #38311: Lattices are missing rcd_act/rcd_vals
Revision: 1c6cd23871567e958cd337aac4a27f5ed805cb84

Lattices are missing rcd_vals/rcd_act for RCD_FLOORWALL mode.

This means that players can't create plating by clicking on lattice with an RCD in 'Floors & Walls mode' - they have to click the space turf beneath.

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
fix: RCDs can now construct floors by clicking on lattices and catwalks (same as floor tiles).
/:cl:

Fixes: #38311

It was always pretty annoying that floor tiles can construct floors by clicking on lattices/catwalks, but RCD users had to click on the space turf underneath.

### CODE REVIEW

File: code/game/objects/structures/lattice.dm [hunk 1]
```diff
@@ -55,6 +55,27 @@
 	if(current_size >= STAGE_FOUR)
 		deconstruct()
 
+/obj/structure/lattice/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	switch(the_rcd.mode)
+		if(RCD_FLOORWALL)
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 1)
```
Comment (81Denton) [inline code/game/objects/structures/lattice.dm]: Cost is the same as using the RCD on a space turf that has a lattice/catwalk placed on top.

Comment (AnturK) [reply code/game/objects/structures/lattice.dm]: Call underlying space rcd_vals then instead of duplicating the values.

Comment (81Denton) [reply code/game/objects/structures/lattice.dm]: How do I call the rcd_vals of /turf/open/space? Sorry, I'm a bit new to this.

Comment (AnturK) [reply code/game/objects/structures/lattice.dm]: Get the turf with `get_turf(lattice)` then just call rcd_vals(user,the_rcd) on it. Make sure it's space, you can pass the action through in the effect call too.

**Comment by 81Denton** [comment]
I sadly don't know how to put what @AnturK wrote into code. Does anyone have a bit of free time to look at it for me?

**Comment by ninjanomnom** [comment]
Get in the irc for help with that sort of thing

**Comment by SpaceManiac** [comment]
In each function:
1. T = get_turf(src)
2. check isspaceturf(T)
3. return T.function(arg, arg)

**Comment by 81Denton** [comment]
Thanks for the offer, I'll reopen this when I'm done with some other things.

**Pull request closed without merging**